### PR TITLE
Verify Task definitions contain a certain result

### PR DESCRIPTION
**Pull request by lcarva**
Ref: EC-794

### CODE REVIEW

File: policy/task/results.rego [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	some error in _rule_data_errors
+	result := lib.result_helper(rego.metadata.chain(), [error])
+}
+
+errors contains err if {
+	some required in lib.rule_data(_rule_data_key)
+	input.metadata.name == required.task
+	found := [result |
+		some result in input.spec.results
+		result.name == required.result
```
Comment (simonbaird) [inline policy/task/results.rego:49]: Probably an unlikely edge case, but should we check that a value key is present as well?

**Review by simonbaird — APPROVED** [review]

File: policy/task/results_test.rego [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	lib.assert_equal_results(results.deny, expected) with input as _mock_task
+		with data.rule_data as d
+}
+
+_mock_task := {
+	"apiVersion": "tekton.dev/v1",
+	"kind": "Task",
+	"metadata": {"name": "bacon"},
+	"spec": {"results": [
+		{"name": "GRILLED"},
```
Comment (simonbaird) [inline policy/task/results_test.rego:82]: The missing value here made me think of it..

File: policy/task/results.rego (hunk 1 above)
Comment (lcarva) [reply policy/task/results.rego:49]: This is verifying a Task definition. There won't be a value. The result would be present in the status of a TaskRun.

File: policy/task/results_test.rego (hunk 2 above)
Comment (simonbaird) [reply policy/task/results_test.rego:82]: (Disregard)

**Pull request merged**